gh-badges/lib/make-badge.js

```diff
@@ -108,75 +22,43 @@ module.exports = function makeBadge({
   // String coercion and whitespace removal.
   text = text.map(value => `${value}`.trim())
 
-  let [left, right] = text
+  const [label, message] = text
 
   color = normalizeColor(color)
   labelColor = normalizeColor(labelColor)
 
   // This ought to be the responsibility of the server, not `makeBadge`.
   if (format === 'json') {
     return JSON.stringify({
-      label: left,
-      message: right,
+      label,
+      message,
       logoWidth,
       color,
       labelColor,
       link: links,
-      name: left,
-      value: right,
+      name: label,
+      value: message,
     })
   }
 
-  if (!(template in templates)) {
-    if (template === 'popout-square') {
-      template = 'flat-square'
-    } else {
-      template = 'flat'
-    }
-  }
-  if (template === 'social') {
-    left = capitalize(left)
-  } else if (template === 'for-the-badge') {
-    left = left.toUpperCase()
-    right = right.toUpperCase()
-  }
-
-  let leftWidth = (anafanafo(left) / 10) | 0
-  // Increase chances of pixel grid alignment.
-  if (leftWidth % 2 === 0) {
-    leftWidth++
-  }
-  let rightWidth = (anafanafo(right) / 10) | 0
-  // Increase chances of pixel grid alignment.
-  if (rightWidth % 2 === 0) {
-    rightWidth++
+  const methodName = camelcase(template)
+  if (!(methodName in badgeRenderers)) {
+    throw new Error(`Unknown template: '${template}'`)
```

Comment:
Not blocking, though at some point this should be consolidated with the style validation logicin gh-badges/lib/index.js.